lightning/config/config.go

```diff
@@ -286,9 +290,47 @@ func (cfg *Config) Load() error {
 	return nil
 }
 
-func (cfg *Config) Adjust() {
+// Adjust fixes the invalid or unspecified settings to reasonable valid values.
+func (cfg *Config) Adjust() error {
 	cfg.App.Config.Adjust()
 
+	// automatically determine the TiDB port & PD address from TiDB settings
+	if cfg.TiDB.Port <= 0 || len(cfg.TiDB.PdAddr) == 0 {
+		resp, err := http.Get(fmt.Sprintf("http://%s:%d/settings", cfg.TiDB.Host, cfg.TiDB.StatusPort))
+		if err != nil {
+			return errors.Annotate(err, "cannot fetch settings from TiDB, please manually fill in `tidb.port` and `tidb.pd-addr`")
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			return errors.Errorf("TiDB settings returned %s, please manually fill in `tidb.port` and `tidb.pd-addr`", resp.Status)
+		}
+		var settings struct {
+			Port             int    `json:"port"`
+			AdvertiseAddress string `json:"advertise-address"`
+			Path             string `json:"path"`
+		}
+		err = json.NewDecoder(resp.Body).Decode(&settings)
+		if err != nil {
+			return errors.Annotate(err, "cannot decode settings from TiDB, please manually fill in `tidb.port` and `tidb.pd-addr`")
+		}
+		if cfg.TiDB.Port <= 0 {
+			_, portStr, err := net.SplitHostPort(settings.AdvertiseAddress)
+			if err == nil {
+				port, err := strconv.Atoi(portStr)
+				if err == nil {
+					cfg.TiDB.Port = port
+				}
+			}
+			if cfg.TiDB.Port <= 0 {
+				cfg.TiDB.Port = settings.Port
+			}
+		}
+		if len(cfg.TiDB.PdAddr) == 0 {
+			pdAddrs := strings.Split(settings.Path, ",")
+			cfg.TiDB.PdAddr = pdAddrs[0] // FIXME support multiple PDs once importer can.
+		}
```

Comment:
check `cfg.TiDB.Port` and `cfg.TiDB.PdAddr` again, if they are empty, throw an error